src/main/java/de/tum/in/www1/artemis/web/rest/hestia/ProgrammingExerciseGitDiffReportResource.java

```diff
@@ -115,4 +126,63 @@ public ResponseEntity<ProgrammingExerciseGitDiffReportDTO> getGitDiffReportForSu
+    @EnforceAtLeastStudent
+    public ResponseEntity<ProgrammingExerciseGitDiffReportDTO> getGitDiffReportForCommitDetailsViewForSubmissions(@PathVariable long exerciseId, @PathVariable long submissionId1,
+            @PathVariable long submissionId2) throws GitAPIException, IOException {
+        log.debug("REST request to get a ProgrammingExerciseGitDiffReport for the commit details view for submission {} and submission {} of exercise {}", submissionId1,
+                submissionId2, exerciseId);
+        var submission1 = submissionRepository.findById(submissionId1).orElseThrow();
+        var submission2 = submissionRepository.findById(submissionId2).orElseThrow();
+        // If either of the two submissions does not belong to the exercise, throw an exception because we do not want to support this for now and while the git diff calucation
+        // would support that,
+        // it would lead to confusing results displayed in the client because the client interface hasn't been designed for this use case.
+        if (!submission1.getParticipation().getExercise().getId().equals(exerciseId) || !submission2.getParticipation().getExercise().getId().equals(exerciseId)) {
+            throw new ConflictException("A git diff report entry can only be retrieved if the submissions exercise id match with the exercise id", ENTITY_NAME,
+                    "exerciseIdsMismatch");
+        }
+        participationAuthCheckService.checkCanAccessParticipationElseThrow(submission1.getParticipation());
+        participationAuthCheckService.checkCanAccessParticipationElseThrow(submission2.getParticipation());
+        var report = gitDiffReportService.generateReportForSubmissions(submission1, submission2);
+        return ResponseEntity.ok(new ProgrammingExerciseGitDiffReportDTO(report));
+    }
+
+    /**
+     * GET exercises/:exerciseId/submissions/:submissionId1/diff-report-commit-details-with-template : Get the diff report for a submission of a programming exercise with the
+     * template of the exercise. To fetch the diff report for the commit details view, the current user needs to have access to the participation of the submission.
+     *
+     * @param exerciseId    the id of the exercise the submission and the template belong to
+     * @param submissionId1 the id of the submission
+     * @return the {@link ResponseEntity} with status {@code 200 (Ok)} and with the diff report as body
+     * @throws GitAPIException if errors occur while accessing the git repository
+     * @throws IOException     if errors occur while accessing the file system
+     */
+    @GetMapping("programming-exercises/{exerciseId}/submissions/{submissionId1}/diff-report-commit-details-with-template")
+    @EnforceAtLeastStudent
+    public ResponseEntity<ProgrammingExerciseGitDiffReportDTO> getGitDiffReportForCommitDetailsViewForSubmissionWithTemplate(@PathVariable long exerciseId,
```

Comment:
Can you explain where and how this endpoint is used?

Reply:
This is used in the Commit Details View, and it is to compare the first commit with the template (which is the previous commit). The template has no submission so i can't use the other method